### PR TITLE
Modernize the build a bit:

### PR DESCRIPTION
**Pull request by genio**
* Update the dzil and build in general
* Use an alienfile
* Use cpanfile for prereqs
* Use DZil ::Starter
* Keep makefile and meta in repo for testing
* Keep license file in repo
* Fixup the changes
* Bump version to 2
* Fix version specific docs
* Switch to README.md
* CI via GH Workflows/Actions
* Add FFI test
* Add XS test
* Don't package upstream libsodium library with the build
* Ensure we're always getting the latest stable version
* Don't do PodWeaver for such a small doc

### CODE REVIEW

File: alienfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+plugin 'PkgConfig' => (
+    pkg_name => 'libsodium',
+    minimum_version => '1.0.3',
+);
+
+share {
+    my $run_upstream_tests = ($ENV{AUTOMATED_TESTING} || $ENV{RELEASE_TESTING}) ? 1 : 0;
+
+    # note on apple weirdisms: https://github.com/joyent/libuv/issues/1200
+    meta->prop->{env}->{LIBTOOLIZE} = 'libtoolize' if $^O eq 'darwin';
```
Comment (plicease) [inline alienfile]: Was this needed generally for macOS? I thought this was a weird-ism of libuv.

File: alienfile [hunk 2]
```diff
@@ -0,0 +1,41 @@
+use alienfile;
+use Config;
+
+configure { requires 'Alien::Build::Plugin::Build::Make' => '0.01' };
+
+plugin 'Probe::Vcpkg' => (
+    name     => 'libsodium',
+    ffi_name => 'libsodium',
```
Comment (plicease) [inline alienfile:8]: nice Visual C++ support.  I can test this next time I'm in front of my  Windows box.

File: alienfile [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        version => qr/^libsodium-([0-9.]+)-stable\.tar\.gz$/,
+    );
+
+    plugin Extract => 'tar.gz';
+
+    requires 'Alien::Autotools';
+    plugin 'Build::Autoconf' => ();
+
+    build [
+        'sh autogen.sh',
```
Comment (plicease) [inline alienfile]: I don't think you need this (or `Alien::Autotools`) as it looks like the tarball comes with a `./configure`.

File: alienfile (hunk 1 above)
Comment (genio) [reply alienfile]: You're right. My glance at a reading through the ticket made me think it was a generic MacOS issue

File: alienfile (hunk 3 above)
Comment (genio) [reply alienfile]: Looks like removing this somehow removed a prereq on Sort::Versions. I'll get that added in specifically.

**Pull request merged**